lib/src/config/object_store.rs

```diff
@@ -195,14 +220,45 @@ fn read_json_contents(file: &Path) -> Result<HashMap<String, String>, ObjectStor
             }
         };
 
-    // Check that each dictionary entry has a string value.
     let mut contents = HashMap::with_capacity(json.len());
+
+    // Check that each KV Store entry has either a string value or an object with data and metadata
+    // fields.
     for (key, value) in json {
-        let value = value
-            .as_str()
-            .ok_or_else(|| ObjectStoreConfigError::FileValueWrongFormat { key: key.clone() })?
-            .to_owned();
-        contents.insert(key, value);
+        let entry = match value {
+            serde_json::Value::String(s) => ObjectStoreEntry {
+                data: s,
+                metadata: None,
+            },
+            serde_json::Value::Object(mut obj) => {
```

Comment:
```suggestion
            serde_json::Value::Object(obj) => {
```
`obj` does not escape this block, so there is no reason to modify it, and thus it does not need to be mutable.